lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php

```diff
@@ -1420,6 +1429,8 @@ public function getInsertSQL()
             $values[] = $placeholder;
         }
 
+        $this->insertSqlColumns = $columns;
```

Comment:
Would it make sense to sort that array?

Reply:
I'm not sure about potential problems but what would be the benefit? If the column order changed in between calls then the current implementation would now work as it kept the column order fixed and if the order does not change than sorting does not have any benefit.

Reply:
If the column order changed then the cache would be reused. That's the benefit.

Reply:
@greg0ire Sure, but that would not work as the column order has to match the given parameter order as the question-mark placeholders are used.

Reply:
This answers my question, thanks!